src/providers/data_providers/businessImageProviderFactory.ts

```diff
@@ -0,0 +1,34 @@
+import { BusinessImageProvider } from './businessImageProvider';
+import { ImageProviderName } from '../../models/data_models/image';
+import { MockBusinessImageProvider } from './mockBusinessImageProvider';
+import { YelpBusinessImageProvider } from './yelpBusinessImageProvider';
+import { Image } from '../../models/data_models/image';
+import { BusinessInfoInput } from '../../models/data_models/types';
+
+export class BusinessImageProviderFactory {
+	private getImageProviderFromString(provider: string): BusinessImageProvider {
+		switch (provider) {
+			case ImageProviderName.mock:
+				return new MockBusinessImageProvider();
+			case ImageProviderName.yelp:
+				return new YelpBusinessImageProvider();
+			default:
+				throw Error(`${provider} is not a valid name for an interpretation provider`);
+		}
+	}
+
+	private getImageProvider(): Array<BusinessImageProvider> {
+		const imageProviders = process.env.IMAGE_PROVIDERS?.split(',').map((provider) => {
+			return this.getImageProviderFromString(provider.trim());
+		});
+		if (imageProviders) return imageProviders;
+		throw Error('IMAGE_PROVIDERS variable not set');
+	}
+
+	public async getImages(businessInfo: BusinessInfoInput): Promise<Array<Image>> {
+		const providers = this.getImageProvider();
+		return await Promise.all(
```

Comment:
```suggestion
		return Promise.all(
```

I think it works without the `await` there?